src/formatDate.js

```diff
@@ -50,7 +50,78 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const currentSep = fromFormat[3];
+  const finalSep = toFormat[3];
+  const dateBefore = date.split(currentSep);
+  const result = [];
+
+  let day;
+  let month;
+  let year;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = dateBefore[i];
+        break;
+
+      case 'MM':
+        month = dateBefore[i];
+        break;
+
+      case 'YY':
+        year = dateBefore[i];
+        break;
+
+      case 'YYYY':
+        year = dateBefore[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (const newFormat of toFormat) {
+    switch (newFormat) {
+      case 'DD':
+        result.push(day);
+        break;
+
+      case 'MM':
+        result.push(month);
+        break;
+
+      case 'YY':
+        result.push(yearFormat(year, newFormat));
+        break;
+
+      case 'YYYY':
+        result.push(yearFormat(year, newFormat));
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return result.join(finalSep);
+}
+
+function yearFormat(year, format) {
+  if (year.length === 4 && format === 'YY') {
+    return year.slice(2);
+  }
+
+  if (year.length === 2 && format === 'YYYY') {
+    if (parseInt(year) < 30) {
+      return '20' + year;
+    } else {
+      return '19' + year;
```

Comment:
```suggestion
return (parseInt(year) < 30)
    ?  '20' + year
    :  '19' +year;
```
i think its better to use ternary situations like this :)

Reply:
Thank you, will do so. 👍